examples/jupyter/tutorials/tutorial6-python-qfunction.ipynb

```diff
@@ -0,0 +1,348 @@
+    "v.set_value(0)\n",
+    "\n",
+    "inputs = [ u ]\n",
+    "outputs = [ v ]\n",
+    "qf.apply(q, inputs, outputs)\n",
+    "\n",
+    "v_array = v.get_array_read()\n",
+    "\n",
+    "print(v_array)\n",
+    "\n",
+    "v.restore_array_read()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "* In the following example, we create and evaluate a built-in identity QFunction with size > 1."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[  0.   1.   4.   9.  16.  25.  36.  49.  64.  81. 100. 121. 144. 169.\n",
+      " 196. 225. 256. 289. 324. 361. 400. 441. 484. 529.]\n"
+     ]
+    }
+   ],
+   "source": [
+    "size = 3\n",
+    "\n",
+    "qf = ceed.IdentityQFunction(size, libceed.EVAL_INTERP, libceed.EVAL_INTERP)\n",
+    "\n",
+    "q = 8\n",
+    "\n",
+    "u_array = np.zeros(q*size, dtype=\"float64\")\n",
+    "for i in range(q*size):\n",
+    "  u_array[i] = i*i\n",
+    "\n",
+    "u = ceed.Vector(q*size)\n",
+    "u.set_array(u_array, cmode=libceed.USE_POINTER)\n",
+    "v = ceed.Vector(q*size)\n",
+    "v.set_value(0)\n",
+    "\n",
+    "inputs = [ u ]\n",
+    "outputs = [ v ]\n",
+    "qf.apply(q, inputs, outputs)\n",
+    "\n",
+    "v_array = v.get_array_read()\n",
+    "\n",
+    "print(v_array)\n",
+    "\n",
+    "v.restore_array_read()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "* In the following example, we create and view two QFunctions (for the setup and apply, respectively, of the mass operator in 1D) from the gallery of available built-in QFunctions in libCEED"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Gallery CeedQFunction Mass1DBuild\n",
+      "  2 Input Fields:\n",
+      "    Input Field [0]:\n",
+      "      Name: \"dx\"\n",
+      "      Size: 1\n",
+      "      EvalMode: \"gradient\"\n",
+      "    Input Field [1]:\n",
+      "      Name: \"weights\"\n",
+      "      Size: 1\n",
+      "      EvalMode: \"quadrature weights\"\n",
+      "  1 Output Field:\n",
+      "    Output Field [0]:\n",
+      "      Name: \"qdata\"\n",
+      "      Size: 1\n",
+      "      EvalMode: \"none\"\n",
+      "\n",
+      "Gallery CeedQFunction MassApply\n",
+      "  2 Input Fields:\n",
+      "    Input Field [0]:\n",
+      "      Name: \"u\"\n",
+      "      Size: 1\n",
+      "      EvalMode: \"interpolation\"\n",
+      "    Input Field [1]:\n",
+      "      Name: \"qdata\"\n",
+      "      Size: 1\n",
+      "      EvalMode: \"none\"\n",
+      "  1 Output Field:\n",
+      "    Output Field [0]:\n",
+      "      Name: \"v\"\n",
+      "      Size: 1\n",
+      "      EvalMode: \"interpolation\"\n",
+      "\n"
```

Comment:
I think this would be very useful at the top. It makes it clearer what each QFunction does